source/docs/casper/dapp-dev-guide/writing-contracts/contracts-and-session-code.md

```diff
@@ -0,0 +1,579 @@
+# Smart Contracts and Session Code
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# 1. Session Code {#session-code}
+Session code is the simplest logic one can execute on a Casper network. It is essential because it is often used to trigger contract logic stored on the chain. Session code requires only one entry point, the `call` function, and it runs within the context of the account executing the session code. As a result, the session code runs with the account's permissions, such as having access to the account's main purse. For example, the session code could transfer tokens from the account's main purse.
+
+The best use of session code is when the situation calls for [stateless](/glossary/S.md/#stateless) execution, and very little or no internal data needs to be tracked. Session code is required when interacting and accepting values returned across the Wasm boundary.
+
+## Comparing Session and Contract Code
+
+The following table summarises the key differences between session code and contract code on a Casper network.
+
+| Session Code | Contract Code |
+| --- | --- |
+| Session code always executes in the context of the account that signed the deploy containing the session code. | A smart contract,  which is stored on-chain logic, executes within its own context. |
+| Session code has only one entry point, `call`, which can be used to interact with the session code. | A smart contract can have multiple entry points that can be invoked.|  
+| The `call` entry point initiates any action the session code takes. | Any action undertaken by a contract must initiate through an outside call, usually via session code.|
+| When a `put_key` call is made within the body of the session code, the key is added to the account's named keys. | When a `put_key` call is made within the smart contract's context, the contract's record is modified to have a new named_key entry. |
+| For more information on how to write session code, see [Writing Session Code](/dapp-dev-guide/writing-contracts/contracts-and-session-code#writing-session-code). | For more information on writing contracts, see [Writing a Basic Smart Contract in Rust](/dapp-dev-guide/writing-contracts/rust-contracts). |
+
+The following image depicts the comparison presented in the table.
+
+<p align="center"><img src={useBaseUrl("/image/dApp/contract-vs-session.png")} alt="Comparing Session and Contract Code" /></p>
+
+**When should you use Session Code?**
+1. When transferring funds from the account's main purse
+2. When configuring multisig thresholds or assigning weights to keys
+3. When you need to call a Smart Contract Entry Point in the account's context
+
+## Video Walkthrough
+
+The following brief video describes [sample session code](https://github.com/casper-ecosystem/two-party-multi-sig/) for configuring an account.
+
```

Comment:
Is there a reason this is set to a fixed width instead of being allowed to scale? If someone wants to watch it in the same window instead of opening YouTube, it's pretty small.

Reply:
To be clear, I am referring to the Youtube embed just below.

Reply:
I copied this from the old doc, will see if I change it to a responsive embed.